modules/ipp/scanner.go

```diff
@@ -224,83 +208,196 @@ func bufferFromBody(res *http.Response, scanner *Scanner) *bytes.Buffer {
 	return b
 }
 
-// FIXME: This will read the wrong section of the body if a substring matches the attribute name passed in
-// TODO: Support reading from multiple instances of the same attribute in a response
-func readAttributeFromBody(attrString string, body *[]byte) ([][]byte, error) {
-	attr := []byte(attrString)
-	interims := bytes.Split(*body, attr)
-	if len(interims) > 1 {
-		valueTag := interims[0][len(interims[0])-3]
-		var vals [][]byte
-		buf := bytes.NewBuffer(interims[1])
-		// This reading occurs in a loop because some attributes can have type "1 setOf <type>"
-		// where same attribute has a set of values, rather than one
-		for tag, nameLength := valueTag, int16(0); tag == valueTag && nameLength == 0; {
-			var length int16
-			if err := binary.Read(buf, binary.BigEndian, &length); err != nil {
-				//Couldn't read length of content
-				return vals, err
+type Value struct {
+	Bytes []byte `json:"raw,omitempty"`
+}
+
+type Attribute struct {
+	Name string    `json:"name,omitempty"`
+	Values []Value `json:"values,omitempty"`
+	ValueTag byte  `json:"tag,omitempty"`
+}
+
+func shouldReturnAttrs(length, soFar, size, upperBound int) (bool, error) {
+	if soFar + length > size {
+		if size == upperBound {
```

Comment:
Am I reading this wrong, or is this backwards? `size == upperBound` would mean that it got truncated, so that would be the non-error case, right?

Reply:
You're definitely correct, I just wrote it backward.